### DISCUSSION TITLE
Using "nonQueued" causes exception.

### DISCUSSION THREAD

**Discussion by marbuser** [spatie/laravel-medialibrary#1859] [Q&A]
I really want to do image conversions on demand because I'm having issues with workers and queues and doing it on demand fits my use-case more anyway.

However whenever I use the `nonQueued()` function on my conversion, I get an exception thrown; `Call to undefined method Symfony\Component\Process\Process::fromShellCommandline()`

It claims it's being thrown from `vendor/spatie/image-optimizer/src/OptimizerChain.php:97`

I have absolutely no idea why this is happening and I feel like I've looked everywhere and not found a solution. I've installed all required dependencies, I've tried swapping between imagick and gq with no difference and same error, so hopefully someone can help me out.

Please let me know if I can provide more information.

**Comment by GregPeden** [comment]
Does it make a difference what you are uploading/submitting? I am aware of a specific edge case with GhostScript whereby the security controls in its default config file effectively block it from being used, and the behavior and setup requirements are different between the queued and non-queued case.

Can you by chance post the full exception call trace here?
